src/ConnectionPool.jl

```diff
@@ -358,115 +383,45 @@ function purge(c::Connection)
     @ensure bytesavailable(c) == 0
 end
 
-"""
-The `pool` is a collection of open `Connection`s.  The `request`
-function calls `getconnection` to retrieve a connection from the
-`pool`.  When the `request` function has written a Request Message
-it calls `closewrite` to signal that the `Connection` can be reused
-for writing (to send the next Request). When the `request` function
-has read the Response Message it calls `closeread` to signal that
-the `Connection` can be reused for reading.
-"""
-const pool = [Vector{Connection}()]
-const poollock = [ReentrantLock()]
-const poolcondition = [Condition()]
-
 """
     closeall()
 
 Close all connections in `pool`.
 """
 function closeall()
-    lock(poollock[Threads.threadid()]) do
-        for c in pool[Threads.threadid()]
-            close(c)
+    for pod in values(POOL.conns)
+        while isready(pod.conns)
+            close(take!(pod.conns))
         end
-        empty!(pool[Threads.threadid()])
+        Threads.atomic_xchg!(pod.numactive, 0)
```

Comment:
weird to use xchg (and atomic at that), but ignore the result. seems like you just mean `pod.numactive[] = 0`?

Reply:
ah yes, I think I was just deep in the Atomic docs and saw this and used it w/o thinking about it more reasonably

Reply:
Updated